File: Problems/No.869_Reordered_Power_of_2/reordered_power_of_2.cpp
class Solution {
public:
    bool reorderedPowerOf2(int n) {
        
        // start computing
        if (n == 1) return true;

        std::string n_str = std::to_string(n);
        sort(n_str.begin(), n_str.end());

        int curr_p2 = 1;
        for (int i=1; i<32; i++) {

            curr_p2 <<= 1;
            std::string curr_p = std::to_string(curr_p2);
            sort(curr_p.begin(), curr_p.end());
            if (n_str == curr_p) return true;

        }

        return false;

    }
};

File: Problems/No.869_Reordered_Power_of_2/reordered_power_of_2.py
class Solution(object):
    def reorderedPowerOf2(self, n):
        """
        :type n: int
        :rtype: bool
        """

        # start computing
        if (n == 1):
            return True
        
        n_str = list(str(n))
        n_str.sort()

        curr_p2 = 1
        for i in range(1, 32):
            curr_p2 <<= 1
            curr_p = list(str(curr_p2))
            curr_p.sort()
            if (n_str == curr_p):
                return True
        
        return False
